flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlToOperationConverter.java

```diff
@@ -179,6 +184,37 @@ private Operation convertDropTable(SqlDropTable sqlDropTable) {
 		return new DropTableOperation(identifier, sqlDropTable.getIfExists());
 	}
 
+	/** convert ALTER TABLE statement. */
+	private Operation convertAlterTable(SqlAlterTable sqlAlterTable) {
+		UnresolvedIdentifier unresolvedIdentifier = UnresolvedIdentifier.of(sqlAlterTable.fullTableName());
+		ObjectIdentifier tableIdentifier = catalogManager.qualifyIdentifier(unresolvedIdentifier);
+		ObjectIdentifier newTableIdentifer = null;
+		Map<String, String> properties = new HashMap<>();
+		CatalogTable catalogTable = null;
+		if (sqlAlterTable.isRename()) {
+			UnresolvedIdentifier newUnresolvedIdentifier = UnresolvedIdentifier.of(sqlAlterTable.fullNewTableName());
+			newTableIdentifer = catalogManager.qualifyIdentifier(newUnresolvedIdentifier);
+		} else {
+			Optional<CatalogManager.TableLookupResult> optionalCatalogTable = catalogManager.getTable(tableIdentifier);
+			if (optionalCatalogTable.isPresent() && !optionalCatalogTable.get().isTemporary()) {
+				CatalogTable originalCatalogTable = (CatalogTable) optionalCatalogTable.get().getTable();
+				properties.putAll(originalCatalogTable.getProperties());
+				sqlAlterTable.getPropertyList().getList().forEach(p ->
+								properties.put(((SqlTableOption) p).getKeyString().toLowerCase(),
```

Comment:
do we force users to have lower case keys?

Reply:
That's a very good question. Now the create table converter will lower the properties too. And I talked with @danny0405  offline and agreed to change the current behavior, so I create a issue to track [FLINK-15061](https://issues.apache.org/jira/browse/FLINK-15061). I'll fix this problem in another PR.